src/System.Device.Gpio/System/Device/I2c/II2cController.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Device.I2c
+{
+    public interface II2cController : IDisposable
+    {
+        /// <summary>
+        /// Determines if this should be disposed.
+        /// </summary>
+        bool ShouldDispose { get; set; }
+
+        /// <summary>
+        /// Opens an I2C device with specified settings in order for it to be ready to use with the controller.
+        /// </summary>
+        /// <param name="settings">The connection settings of a device on an I2C bus.</param>
+        /// <param name="shouldDispose">Determines if this should be disposed.</param>
+        void OpenDevice(I2cConnectionSettings settings, bool shouldDispose = true);
```

Comment:
Is this optional parameter supposed to be set in the interface?  Was not sure if that is the norm.

Reply:
Not included any longer.